### ISSUE TITLE
Feature request: block message removals

### ISSUE THREAD

**Issue by justanotheranonymoususer** [tomer8007/whatsapp-web-incognito#31]
It's not exactly related to being incognito, but I think that it can be a nice addition, and perhaps it's easy to do with the toolset you already have. When somebody deletes a message, I believe it's possible to block the request on the browser and leave it at least until the page is refreshed. What do you think?

**Comment by tomer8007** [comment]
Interesting idea, it's just not so obvious that all users would want it

**Comment by justanotheranonymoususer** [comment]
Can be made opt-in I guess

**Comment by tomer8007** [comment]
But that would make the UI uglier

**Comment by justanotheranonymoususer** [comment]
Looks great

![image](https://user-images.githubusercontent.com/5781692/97114929-18e28480-16fc-11eb-8e00-9e1a6dbe22e6.png)

**Comment by tomer8007** [comment]
Hmm, I'm not sure, what do you about making this feature non opt-in (always enabled)?

**Comment by justanotheranonymoususer** [comment]
I'd be using it anyway, so I don't mind. But sometimes there might be value in allowing to disable selected features. In this case, maybe it can break WA one day. It's up to you, I'd be happy to get this feature in any form and with any UI :)

**Comment by tomer8007** [comment]
Alright, I'll implment this sometime

**Comment by tomer8007** [comment]
OK, so it appears that message deletions are implemented as packets of type `modify_tag`, and then the reception of another message (I guess that is the "This message was deleted" text.)

Not sure how blocking will be done based on that, but right now the extension does not have the abillity to manipulate incoming packets, so I'm making this possible first.

**Comment by justanotheranonymoususer** [comment]
Thanks for the update!

P.S.1. There's [something remotely similar](https://github.com/SpriteOvO/Telegram-Anti-Revoke) for telegram. Quite different tech but same goal, perhaps you're interested to know.

P.S.2. Does whatsapp-web-incognito block online status and typing...? If not, I can create feature request issues for these too :)

**Comment by tomer8007** [comment]
It blocks presense updates (I guess that's what you mean by online status?) and "composing" events (typing). It also blocks read receipts for statuses. I recommend checking that yourself, and if that doesn't work for some reason let me know.

**Comment by justanotheranonymoususer** [comment]
I checked the privacy stuff.

> It blocks presence updates

True. Even though the checkbox was unchecked by itself on my computer, not sure why. Perhaps after an update. I checked it again and now looks like it's good.

> "composing" events (typing)

Also, very nice.

> Message removal blocking was implemented

Confirmed too, works nicely, thanks! A small nice addition can be to add an 🚫 icon or something near the message to mark it as removed, to get some indication and know to back it up or something if relevant.
